api/client.go

```diff
@@ -0,0 +1,142 @@
+// Package api used for communication with the KSeF API
+package api
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"os"
+
+	"github.com/go-resty/resty/v2"
+)
+
+// ClientOptFunc defines function for customizing the KSeF client
+type ClientOptFunc func(*ClientOpts)
+
+// ClientOpts defines the client parameters
+type ClientOpts struct {
+	Client           *resty.Client
+	URL              string
+	ID               string
+	Token            string
+	SessionToken     string
+	SessionReference string
+	KeyPath          string
+}
+
+func defaultClientOpts() ClientOpts {
+	return ClientOpts{
+		Client:           resty.New(),
+		URL:              "https://ksef-test.mf.gov.pl",
```

Comment:
I'm thinking about preparing 2 configurations. One for the sandbox (ksef-test.mf.gov.pl) and the other for production (ksef.mf.gov.pl). Does this make sense? The user of the package would still have to specify what environment wants to use.

Reply:
This is the suggested go way of doing things. You have opts and you can configure the client to your needs by passing in a parameter for the env you want.
```
ksef_api.NewClient(
		ksef_api.WithProductionURL()
	)
```
By default we set the test so that we don't make any request to production by mistake, but you can always switch to production or demo. The CLI at the moment uses only the test API. I was on the fence on whether I should have a CLI command to send stuff to production. The command itself already takes in a lot of arguments so I was thinking about implementing it as a separate command, but decided to hold back on it for now.